service/domain/feeds/contact.go

```diff
@@ -0,0 +1,78 @@
+package feeds
+
+import (
+	"fmt"
+
+	"github.com/boreq/errors"
+	"github.com/planetary-social/scuttlego/service/domain/feeds/content"
+	"github.com/planetary-social/scuttlego/service/domain/refs"
+)
+
+type Contact struct {
+	target    refs.Identity
+	following bool
+	blocking  bool
+}
+
+func NewContact(target refs.Identity) (*Contact, error) {
+	if target.IsZero() {
+		return nil, errors.New("zero value of target")
+	}
+	return &Contact{
+		target: target,
+	}, nil
+}
+
+func NewContactFromHistory(target refs.Identity, following, blocking bool) (*Contact, error) {
```

Comment:
I'd suggest making `following` and `blocking` strongly typed as now it's easy to mix them up while passing the arguments.

Reply:
Will this actually eliminate any mistakes if this function is directly called in the adapters and we just pass bools to it? 

https://github.com/planetary-social/scuttlego/blob/correctly-update-contacts/service/adapters/bolt/social_graph_repository.go#L112

Reply:
Note: `c.Following` and `c.Blocking` will always be raw bools in this scenario as this is a transport type/type used for serialisation or whatever you want to call it.